atd-etl/app/process_socrata_export.py

```diff
@@ -6,11 +6,99 @@
 Description: The purpose of this script is to gather data from Hasura
 and export it to the Socrata database.
 
-The application requires the requests library:
+The application requires the requests and sodapy libraries:
     https://pypi.org/project/requests/
+    https://pypi.org/project/sodapy/
 """
-
+import os
+import time
+from string import Template
+from sodapy import Socrata
 from process.config import ATD_ETL_CONFIG
-print("Socrata - Exporter:  Not yet implemented.")
-print("SOCRATA_KEY_ID: " + ATD_ETL_CONFIG["SOCRATA_KEY_ID"])
-print("SOCRATA_KEY_SECRET: " + ATD_ETL_CONFIG["SOCRATA_KEY_SECRET"])
+from process.helpers_socrata import *
+from process.socrata_queries import *
+print("Socrata - Exporter:  Started.")
+
+# Setup connection to Socrata
+client = Socrata("data.austintexas.gov", ATD_ETL_CONFIG["SOCRATA_APP_TOKEN"],
+                 username=ATD_ETL_CONFIG["SOCRATA_KEY_ID"], password=ATD_ETL_CONFIG["SOCRATA_KEY_SECRET"], timeout=20)
+
+# Define tables to query from Hasura and publish to Socrata
+query_configs = [
+    {
+        "table": "crash",
+        "template": crashes_query_template,
+        "formatter": format_crash_data,
+        "formatter_config": {
+            "tables": ["atd_txdot_crashes"],
+            "columns_to_rename": {
+                "veh_body_styl_desc": "unit_desc",
+                "veh_unit_desc_desc": "unit_mode",
+            },
+            "flags_list": ["MOTOR VEHICLE",
```

Comment:
I like the way this is implemented. that is all.

Reply:
Thanks! I was pretty happy that I was able refactor this way. I was hoping to come up with one formatter for any table type but that could come down the road with other datasets  that we publish.